src/plugins/management/public/types.ts

```diff
@@ -17,6 +17,82 @@
  * under the License.
  */
 
+import { IconType } from '@elastic/eui';
+import { ManagementApp } from './management_app';
+import { ManagementSection } from './management_section';
+import { ChromeBreadcrumb, ApplicationStart } from '../../../core/public/';
+
+export interface ManagementSetup {
+  sections: SectionsServiceSetup;
+}
+
 export interface ManagementStart {
+  sections: SectionsServiceStart;
   legacy: any;
 }
+
+interface SectionsServiceSetup {
+  getSection: (sectionId: ManagementSection['id']) => ManagementSection | undefined;
+  getAllSections: () => ManagementSection[];
+  register: RegisterSection;
+}
+
+interface SectionsServiceStart {
+  getSection: (sectionId: ManagementSection['id']) => ManagementSection | undefined;
+  getAllSections: () => ManagementSection[];
+  navigateToApp: ApplicationStart['navigateToApp'];
+}
+
+export interface CreateSection {
+  id: string;
+  title: string;
+  order?: number;
+  euiIconType?: string; // takes precedence over `icon` property.
+  icon?: string; // URL to image file; fallback if no `euiIconType`
+}
+
+export type RegisterSection = (section: CreateSection) => ManagementSection;
+
+export interface RegisterManagementAppArgs {
+  id: string;
+  title: string;
+  mount: ManagementSectionMount;
+  order?: number;
+}
+
+export type RegisterManagementApp = (managementApp: RegisterManagementAppArgs) => ManagementApp;
+
+export type Unmount = () => Promise<void> | void;
+
+interface ManagementAppMountParams {
+  basePath: string; // base path for setting up your router
+  element: HTMLElement; // element the section should render into
+  setBreadcrumbs: (crumbs: ChromeBreadcrumb[]) => void;
+}
+
+export type ManagementSectionMount = (
+  params: ManagementAppMountParams
```

Comment:
Perhaps we should have a conversation with @joshdover about this. I've been confused about the purpose of context containers and I'm therefore uncertain of anything we're discussing in relationship to them.

Let's discuss via slack or zoom since I think we should even up on needs / expectations before talking interfaces.

Reply:
spoke with @streamich and his questions are mainly regarding what the platform team is providing rather than this implementation.